modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/PointStackerProcess.java

```diff
@@ -201,30 +289,40 @@ public SimpleFeatureCollection execute(
             weightClusterPosition = argWeightClusterPosition;
         }
 
+        if (weightClusterPosition) {
+            clusterType = ClusterType.GridWeight;
+        }
+
+        boolean computeBBox = false;
+        if (argComputeBBox != null) {
+            computeBBox = argComputeBBox;
+        }
+
+        // default value
+        if (clusterType == null) {
+            clusterType = ClusterType.GridCenter;
+        }
         // TODO: allow output CRS to be different to data CRS
         // assume same CRS for now...
         double cellSizeSrc = cellSize * outputEnv.getWidth() / outputWidth;
 
-        // create cluster points, based on cellSize and width and height of the viewd area.
         Collection<StackedPoint> stackedPts =
                 stackPoints(
                         data,
                         crsTransform,
                         cellSizeSrc,
-                        weightClusterPosition,
                         outputEnv.getMinX(),
-                        outputEnv.getMinY());
+                        outputEnv.getMinY(),
+                        clusterType,
+                        returnClusteredAttribute);
 
         SimpleFeatureType schema = createType(srcCRS, normalize);
         ListFeatureCollection result = new ListFeatureCollection(schema);
         SimpleFeatureBuilder fb = new SimpleFeatureBuilder(schema);
 
-        GeometryFactory factory = new GeometryFactory(new PackedCoordinateSequenceFactory());
-
+        GeometryFactory factory = new GeometryFactory();
```

Comment:
I had an older version of the library that was using vividsolutions, when I changed to locationtech that slipped along the crack, I'll get back to it. thank you.

Reply:
Ok so this would collect all the values in an array... that's the bit that's really unfortunate here then, much more so than the name. If the group is large (happens when zooming out) then this array can become very large, and cause an OOM.
OOMs on servers are pretty bad, because they can take down the other requests running in parallel as well. 
Can you put some protection, like for example a max number of values collected in the array (e..g, 10000) and throw an exception if they go beyond? The default value could be tunable via a system property.

Also, the description of the parameter should say that all values are collected in an array, and returned.

Reply:
Agreed. I shall put a restriction and I'll make it configurable via a property which will be initialized with 10000 but can be overridden.